File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/ServiceRegistry.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vsf;

import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsProxy;
import org.mozilla.mod.javascript.Scriptable;
import org.eclipse.vjet.dsf.javatojs.anno.AExclude;
import org.eclipse.vjet.dsf.dap.proxy.INativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsFuncProxy;
import org.eclipse.vjet.dsf.dap.proxy.NativeJsTypeRef;
import org.eclipse.vjet.dsf.javatojs.anno.AJavaOnly;
import org.eclipse.vjet.vsf.IBinding;
import org.eclipse.vjet.vsf.ITransport;

//NativeJsProxy for vjo.dsf.ServiceRegistry.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public class ServiceRegistry extends NativeJsProxy {

    /** for framework use only */
    @AExclude
    public ServiceRegistry(Scriptable nativeObj){
        super(nativeObj);
    }

    /** internal use only */
    protected ServiceRegistry(Object ...args){
        super(args);
    }

    public ServiceRegistry() {
        super();
    }

    public static IBinding getBinding(String payloadType) {
        return callStaticWithName("vjo.dsf.ServiceRegistry", "getBinding", IBinding.class, payloadType);
    }

    public static ITransport getTransport(String trspType) {
        return callStaticWithName("vjo.dsf.ServiceRegistry", "getTransport", ITransport.class, trspType);
    }

    public static boolean registerBinding(String payloadType, IBinding binding) {
        return callStaticWithName("vjo.dsf.ServiceRegistry", "registerBinding", Boolean.class, payloadType, binding);
    }

    public static boolean registerTransport(String trspType, ITransport transport) {
        return callStaticWithName("vjo.dsf.ServiceRegistry", "registerTransport", Boolean.class, trspType, transport);
    }

    @AJavaOnly
    public static final NativeJsTypeRef<ServiceRegistry> prototype = NativeJsTypeRef.get(ServiceRegistry.class);

    public static final INativeJsFuncProxy<NativeJsTypeRef<ServiceRegistry>> getBinding = NativeJsFuncProxy.create(prototype, "getBinding");

    public static final INativeJsFuncProxy<NativeJsTypeRef<ServiceRegistry>> getTransport = NativeJsFuncProxy.create(prototype, "getTransport");

    public static final INativeJsFuncProxy<NativeJsTypeRef<ServiceRegistry>> registerBinding = NativeJsFuncProxy.create(prototype, "registerBinding");

    public static final INativeJsFuncProxy<NativeJsTypeRef<ServiceRegistry>> registerTransport = NativeJsFuncProxy.create(prototype, "registerTransport");
}

File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/ServiceRegistry.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
vjo.ctype('org.eclipse.vjet.vsf.ServiceRegistry') //< public
.needs('org.eclipse.vjet.vsf.ITransport')
//> needs org.eclipse.vjet.vsf.IBinding
.props({
	// transports
	t : [], //< private static ITransport[]
	// bindings
	b : [], //< private static IBinding[]
	//> public IBinding getBinding(String payloadType);
	getBinding : function(payloadType) {
		return this.b[payloadType]; //< IBinding
	},
	//> public ITransport getTransport(String trspType);
	getTransport : function(trspType) {
		return this.t[trspType];
	},
	/**
	 * returns true if registered else false
	 */
	//> public boolean registerBinding(String payloadType, IBinding binding);
	registerBinding : function(payloadType, binding) {
		if (payloadType && binding) {
			this.b[payloadType] = binding;
			return true;
		}
		return false;
	},
	/**
	 * returns true if registered else false
	 */
	//> public boolean registerTransport(String trspType, ITransport transport);
	registerTransport : function(trspType, transport) {
		if (trspType && transport) {
			if (vjo.isInstanceOf(transport, org.eclipse.vjet.vsf.ITransport)) {
				// only register if trspType is not already registered
				if (null == this.t[trspType]) {
					this.t[trspType] = transport;
					return true;
				}
			}
		}
		return false;
	}
})
.endType();